torch/csrc/jit/profiling_graph_executor_impl.cpp

```diff
@@ -178,6 +178,29 @@ ExecutionPlan ProfilingGraphExecutorImpl::getPlanFor(
   return *optimized_plan_;
 }
 
+static void removeProfileNodes(Block* b) {
+  for (auto it = b->nodes().begin(); it != b->nodes().end(); it++) {
+    if (it->kind() == prim::profile) {
+      if (it->outputs().size() == 1) {
+        it->input(0)->setType(it->output()->type());
+        it->output()->replaceAllUsesWith(it->input(0));
+      }
+      it.destroyCurrent();
```

Comment:
Is it safe to destroy nodes in this loop?

Reply:
Yes, destroyCurrent works fine with iterators.